src/NamespaceRepository.php

```diff
@@ -15,6 +15,7 @@
 use WSS\Log\ArchiveSpaceLog;
 use WSS\Log\UnarchiveSpaceLog;
 use WSS\Log\UpdateSpaceLog;
+use Wikimedia\Rdbms\SelectQueryBuilder;
```

Comment:
This import was added, but there are no other changes in the file. Is it needed?

Reply:
I got this:

```
[97105b418a45161246d2b10b] /index.php/Special:AddSpace Error from line 64 of /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/NamespaceRepository.php: Class "WSS\SelectQueryBuilder" not found

Backtrace:

#0 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/NamespaceRepository.php(234): WSS\NamespaceRepository::getNextAvailableNamespaceId()
#1 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/SubmitCallback/AddSpaceSubmitCallback.php(59): WSS\NamespaceRepository->addSpace(WSS\Space)
#2 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(707): WSS\SubmitCallback\AddSpaceSubmitCallback->onSubmit(array, OOUIHTMLForm)
#3 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(597): HTMLForm->trySubmit()
#4 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(613): HTMLForm->tryAuthorizedSubmit()
#5 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/Form/AbstractForm.php(151): HTMLForm->show()
#6 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/UI/AddSpaceUI.php(27): WSS\Form\AbstractForm->show()
#7 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/UI/WSSUI.php(185): WSS\UI\AddSpaceUI->render()
#8 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/Special/SpecialAddSpace.php(65): WSS\UI\WSSUI->execute()
#9 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/SpecialPage.php(24): WSS\Special\SpecialAddSpace->doExecute(string)
#10 /Users/wgevaert/test_wiki/core/includes/specialpage/SpecialPage.php(600): WSS\SpecialPage->execute(string)
#11 /Users/wgevaert/test_wiki/core/includes/specialpage/SpecialPageFactory.php(635): SpecialPage->run(NULL)
#12 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(307): MediaWiki\SpecialPage\SpecialPageFactory->executePath(Title, RequestContext)
#13 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(947): MediaWiki->performRequest()
#14 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(547): MediaWiki->main()
#15 /Users/wgevaert/test_wiki/core/index.php(53): MediaWiki->run()
#16 /Users/wgevaert/test_wiki/core/index.php(46): wfIndexMain()
#17 {main}
```

That being said, after adding the import, on a 1.35 wiki I then got this:
```
[3604e605191cfffa209598a8] /index.php/Special:AddSpace Error from line 65 of /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/NamespaceRepository.php: Undefined constant Wikimedia\Rdbms\SelectQueryBuilder::SORT_DESC

Backtrace:

#0 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/NamespaceRepository.php(235): WSS\NamespaceRepository::getNextAvailableNamespaceId()
#1 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/SubmitCallback/AddSpaceSubmitCallback.php(59): WSS\NamespaceRepository->addSpace(WSS\Space)
#2 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(707): WSS\SubmitCallback\AddSpaceSubmitCallback->onSubmit(array, OOUIHTMLForm)
#3 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(597): HTMLForm->trySubmit()
#4 /Users/wgevaert/test_wiki/core/includes/htmlform/HTMLForm.php(613): HTMLForm->tryAuthorizedSubmit()
#5 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/Form/AbstractForm.php(151): HTMLForm->show()
#6 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/UI/AddSpaceUI.php(27): WSS\Form\AbstractForm->show()
#7 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/UI/WSSUI.php(185): WSS\UI\AddSpaceUI->render()
#8 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/Special/SpecialAddSpace.php(65): WSS\UI\WSSUI->execute()
#9 /Users/wgevaert/test_wiki/core/extensions/WSSpaces/src/SpecialPage.php(24): WSS\Special\SpecialAddSpace->doExecute(string)
#10 /Users/wgevaert/test_wiki/core/includes/specialpage/SpecialPage.php(600): WSS\SpecialPage->execute(string)
#11 /Users/wgevaert/test_wiki/core/includes/specialpage/SpecialPageFactory.php(635): SpecialPage->run(NULL)
#12 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(307): MediaWiki\SpecialPage\SpecialPageFactory->executePath(Title, RequestContext)
#13 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(947): MediaWiki->performRequest()
#14 /Users/wgevaert/test_wiki/core/includes/MediaWiki.php(547): MediaWiki->main()
#15 /Users/wgevaert/test_wiki/core/index.php(53): MediaWiki->run()
#16 /Users/wgevaert/test_wiki/core/index.php(46): wfIndexMain()
#17 {main}
```

To me it seems a bit weird that the SelectQueryBuilder itself is a 1.35 feature, but the `SORT_*` constants are only 1.36+.

I have now changed the `SelectQueryBuilder::SORT_DESC` to `'DESC'` because this extension supports 1.35 per its extension.json

On 1.39 I did not get the second error.